internal/kafka/internal/api/public/model_enterprise_osd_cluster_payload.go

```diff
@@ -16,8 +16,6 @@ type EnterpriseOsdClusterPayload struct {
 	AccessKafkasViaPrivateNetwork bool `json:"access_kafkas_via_private_network"`
 	// The data plane cluster ID. This is the ID of the cluster obtained from OpenShift Cluster Manager (OCM) API
 	ClusterId string `json:"cluster_id"`
```

Comment:
Does this mean enterprise payloads are only supported on OCM?

Reply:
Yes, this is the case at the moment. Enterprise is built on top of ocm. 

But with what's done in this PR, it should be possible to extend to all other cluster providers in the future - e.g a standalone provider for testing or crc clusters etc, by providing the provider type in the request payload. That's not the immediate target for now

Reply:
Setting the expectations on this is relevant.
For example, I see in the line I referenced that ClusterId is "the ID of the cluster obtained from OpenShift Cluster Manager (OCM) API". This basically limits that endpoint to OCM

Reply:
That line makes sense with the current state as OCM is the only supported provider.
But in the likely event when there is another provider implementation hooked in, it'll be revisited - either the documentation is updated or the field (this one or other sibling fields) is completely made optional depending on the underlying providers.